zk.js/src/transaction/pspTransaction.ts

```diff
@@ -328,12 +353,1129 @@ export function createProofInputs({
+            assetPubkeysCircuit[assetIndex]
+          );
+        })
+        .reduce((sum, utxo) => sum.add(utxo.amounts[assetIndex]), new BN(0)),
+    )
+    .add(FIELD_SIZE)
+    .mod(FIELD_SIZE);
+}
+
+/**
+ * Computes the integrity Poseidon hash over transaction inputs that are not part of
+ * the proof, but are included to prevent the relayer from changing any input of the
+ * transaction.
+ *
+ * The hash is computed over the following inputs in the given order:
+ * 1. Recipient SPL Account
+ * 2. Recipient Solana Account
+ * 3. Relayer Public Key
+ * 4. Relayer Fee
+ * 5. Encrypted UTXOs (limited to 512 bytes)
+ *
+ * @param {any} poseidon - Poseidon hash function instance.
+ * @returns {Promise<BN>} A promise that resolves to the computed transaction integrity hash.
+ * @throws {TransactionError} Throws an error if the relayer, recipient SPL or Solana accounts,
+ * relayer fee, or encrypted UTXOs are undefined, or if the encryption of UTXOs fails.
+ *
+ * @example
+ * const integrityHash = await getTxIntegrityHash(poseidonInstance);
+ */
+export function getTxIntegrityHash(
+  relayerFee: BN,
+  encryptedUtxos: Uint8Array,
+  accounts: UnshieldAccounts,
+  verifierConfig: VerifierConfig,
+  verifierProgramId: PublicKey,
+  message?: Uint8Array,
+): BN {
+  if (!relayerFee)
+    throw new TransactionError(
+      TransactionErrorCode.RELAYER_UNDEFINED,
+      "getTxIntegrityHash",
+      "",
+    );
+  if (!accounts.recipientSpl)
+    throw new TransactionError(
+      TransactionErrorCode.SPL_RECIPIENT_UNDEFINED,
+      "getTxIntegrityHash",
+      "",
+    );
+  if (!accounts.recipientSol)
+    throw new TransactionError(
+      TransactionErrorCode.SOL_RECIPIENT_UNDEFINED,
+      "getTxIntegrityHash",
+      "",
+    );
+  if (encryptedUtxos && encryptedUtxos.length > 128 * verifierConfig.out)
+    throw new TransactionParametersError(
+      TransactionParametersErrorCode.ENCRYPTED_UTXOS_TOO_LONG,
+      "getTxIntegrityHash",
+      `Encrypted utxos are too long: ${encryptedUtxos.length} > ${
+        128 * verifierConfig.out
+      }`,
+    );
+
+  const messageHash = message ? sha256(message) : new Uint8Array(32);
+
+  // TODO(vadorovsky): Try to get rid of this hack during Verifier class
+  // refactoring / removal
+  // For example, we could derive which accounts exist in the IDL of the
+  // verifier program method.
+  const recipientSpl =
+    verifierProgramId.toBase58() === lightPsp2in2outStorageId.toBase58()
+      ? new Uint8Array(32)
+      : accounts.recipientSpl.toBytes();
+  // console.log("messageHash ", messageHash);
+  // console.log("recipientSpl ", Array.from(recipientSpl));
+  // console.log("accounts.recipientSol ", Array.from(accounts.recipientSol.toBytes()));
+  // console.log("accounts.relayerPublicKey ", Array.from(accounts.relayerPublicKey.toBytes()));
+  // console.log("relayerFee ", Array.from(relayerFee.toArrayLike(Buffer, "be", 8)));
+  // console.log("encryptedUtxos ", Array.from(encryptedUtxos));
```

Comment:
can we remove these commented logs?

Reply:
Left them on purpose for future debugging of the integrity hash as the otherside to the logs in verifier sdk.